includes/apple-exporter/builders/class-components.php

```diff
@@ -788,6 +789,13 @@ private function split_into_components() {
 		$this->anchor_components( $components );
 		$this->add_pullquote_if_needed( $components );
 
+		$theme          = Theme::get_used();
+		$json_templates = $theme->get_value( 'json_templates' );
+
+		if ( ! empty( $json_templates['end_of_article']['json'] ) ) {
```

Comment:
Can you see if this works if you remove the conditional and just add the component? That is, remove lines 792 & 793 above, and remove this if conditional, and always add the end of article component, and see if the tests still pass. If the component doesn't have any JSON, I think it will just get skipped, and we don't need this logic here. But, if the tests don't pass, then we know we *do* need this.

Reply:
There's an undefined index of 'role' when parsing the components - so I'm going to leave it in for now.